src/document/doc-printer.js

```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-sparse-arrays */
```

Comment:
I was told that [`HOLEY` array](https://v8.dev/blog/elements-kinds#packed-vs.-holey-kinds) are slower, and [v8 says we should avoid creating holes](https://v8.dev/blog/elements-kinds#avoid-creating-holes), but it's been a while. It will be nice if you can run some performance tests.

Reply:
I guess we can just change the order of elements in commands, so that `ind` will be last.
BTW, looks like it'll be better if this targets the `next` branch instead of `main`, right?

Reply:
> I guess we can just change the order of elements in commands, so that ind will be last.

Is it a good idea to use object instead of array?

> BTW, looks like it'll be better if this PR targets the next branch instead of main, right?

Yes, but that won't be a problem, I can rebase, do whatever you like.

Reply:
>Is it a good idea to use object instead of array?

I think so.